test/nginx_system_test.sh

```diff
@@ -633,21 +617,45 @@ function wait_for_new_worker() {
     sleep .1
   done
 }
-wait_for_new_worker
-check wget $EXAMPLE_ROOT/styles/W.rewrite_css_images.css.pagespeed.cf.Hash.css \
-  -O /dev/null
-if [ "$AB_PID" != "0" ]; then
-    echo "Kill ab (pid: $AB_PID)"
-    kill -s KILL $AB_PID &>/dev/null || true
-fi
 
-# There should still be just one babysitter and controller running.
-check_process_names
+if [ "$RUN_CONTROLLER_TEST" = "on" ]; then
+  function check_process_names() {
+    if ! $USE_VALGRIND; then
+      # There should be one babysitter and controller running.  Under valgrind
+      # process labels are confused, so skip the check then.
+
+      running=$(ps auxww | grep 'ngin[x]')
+      check_from "$running" find_exactly_once "nginx: pagespeed babysitter"
+      check_from "$running" find_exactly_once "nginx: pagespeed controller"
+    fi
+  }
+
+  check_process_names
 
-check grep "Writing a byte to a pipe to tell the old controller to exit." \
-  $ERROR_LOG
-check grep "Root process is starting a new controller; shutting down." \
-  $ERROR_LOG
+  # Fire up some heavy load if ab is available to test a stressed reload.
+  # TODO(oschaaf): make sure we wait for the new worker to get ready to accept
+  # requests.
+  fire_ab_load
+
+  URL="$EXAMPLE_ROOT/styles/W.rewrite_css_images.css.pagespeed.cf.Hash.css"
+  check wget "$URL" -O /dev/null
+  check_simple "$NGINX_EXECUTABLE" -s reload -c "$PAGESPEED_CONF"
+
```

Comment:
We want to send ab load and restart nginx regardless, we just shouldn't check controller things if we're not testing the controller.

(Currently this PR disables nginx's entire restart test.)